### DISCUSSION TITLE
Running socioboard with Docker, unable to login

### DISCUSSION THREAD

**Discussion by rancortooth** [socioboard/Socioboard-5.0#393] [Q&A]
I'm trying to get socioboard running in docker as per the instructions in the Socioboard-5.0 repo. I'm able to create a user and then verify it manually via the methods given. I also created a Twilio account and supplied those settings.
The instructions seem to work fine, with the following possible exceptions:

1) I'm not 100% if I entered the expected Twilio values from my Twilio account and the socioboard service I created there. The naming is a bit different between what socioboard specifies and how Twilio refers to these things on their console

2) Due to an update with Docker, I got the following error:
```
ERROR: Network scb-net declared as external but specifies additional attributes (driver, ipam).
```
The fix for this was to comment out the ipam info at the bottom of the docker-compose file. Afterward, everything seemed to build and run fine.
```
    # ipam:
    #   config:
    #     - subnet: 172.16.32.0/16
    #       gateway: 172.16.32.1
```

When I try to login with that user, I keep seeing this error in the User service:
```
/home/socioboard/.pm2/logs/user.server-error.log last 1000 lines:
0|user.ser | TypeError: Cannot read properties of null (reading 'account_count')
0|user.ser |     at default (file:///usr/socioboard/app/socioboard-api/User/middleware/authentication.middleware.js:31:71)
0|user.ser |     at Layer.handle [as handle_request] (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/layer.js:95:5)
0|user.ser |     at trim_prefix (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:328:13)
0|user.ser |     at /usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:286:9
0|user.ser |     at Function.process_params (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:346:12)
0|user.ser |     at next (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:280:10)
0|user.ser |     at /usr/socioboard/app/socioboard-api/User/node_modules/express/lib/application.js:240:9
0|user.ser |     at /usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:646:15
0|user.ser |     at next (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:265:14)
0|user.ser |     at expressInit (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/middleware/init.js:40:5)
0|user.ser |     at Layer.handle [as handle_request] (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/layer.js:95:5)
0|user.ser |     at trim_prefix (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:328:13)
0|user.ser |     at /usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:286:9
0|user.ser |     at Function.process_params (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:346:12)
0|user.ser |     at next (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:280:10)
0|user.ser |     at query (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/middleware/query.js:45:5)
0|user.ser |     at Layer.handle [as handle_request] (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/layer.js:95:5)
0|user.ser |     at trim_prefix (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:328:13)
0|user.ser |     at /usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:286:9
0|user.ser |     at Function.process_params (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:346:12)
0|user.ser |     at next (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:280:10)
0|user.ser |     at Function.handle (/usr/socioboard/app/socioboard-api/User/node_modules/express/lib/router/index.js:175:3)
```

**Comment by vaughngx4** [comment]
Hi, please use my fork for Docker, there are unfixed bugs in the main repo.

**Reply by rancortooth** [reply]
Thank you! I'm using your fork and for some reason I keep getting an error of "Undefined" when I try and create a user. The user tables are blank, so the user isn't being created. I've been going over all the logs I can find but I'm not finding any helpful error messages. There's surprisingly very little in the logs (even under /home/node/.pm2/logs on the socioboard node). I may have inadvertently broken things by changing the domain of "mydomain.example" to "localhost" in the .env file. I just wasn't able to reach the site as "https://socio.mydomain.example", the browser was unable to resolve that DNS name. Using "https://socio.localhost" gets me to the login page and to the page where I can create a user but then I hit the "Undefined" error. I'll keep digging.

**Reply by vaughngx4** [reply]
As of the addition of nginx to the stack, running on localhost is no longer supported and an external DNS resolver is required(see `Docker.README.md`). The "undefined" error is shown when your browser can't reach the api.

If you wish to run it on your local machine, you will need to expose ports `80` and `443` as well as have all domains(see `docker/nginx/ssl/SSL.README.md`) pointing to the IP of your local machine(i.e 192.168.1.27).

**Comment by MacGeyer** [comment]
Same problem here, I have used @vaughngx4's fork, updated domain name, set my Twilio account and set all DNS records, when I try to create an account I get the "undefined" error message

**Reply by vaughngx4** [reply]
This means your api cannot be reach by the frontend, their are several subdomains you need to set up on your network. See the SSL ReadMe. You may also want to see #399 .
